returnn/vocabulary.py

```diff
@@ -21,11 +22,19 @@ class ReturnnVocabFromPhonemeInventory(Job):
     tk.Variable out_vocab_size: integer variable containing the vocabulary size (``vocab_size``)
     """
 
-    def __init__(self, bliss_lexicon: tk.Path):
+    __sis_hash_exclude__ = {"blacklist": None}
+
+    def __init__(
+        self,
+        bliss_lexicon: tk.Path,
+        blacklist: Optional[Union[Iterable, tk.Path]] = None,
```

Comment:
`Iterable` allows `dict`. Is this possibility covered in the code?

Reply:
from code the phonemes need to be given as the values?? maybe extend doc string?

Reply:
Maybe this should be set to `Set` instead, was not sure whether something more generic would be nice, but I am not sure how the code behaves in these situations.
I think they would need to be given as keys.

Reply:
> I think they would need to be given as keys

yes, this is correct.
```
>>> a = {"foo": "bar"}
>>> "foo" in a
True
>>> "bar" in a
False
```

I think `Iterable` would be fine.. if someone inputs a dict as a blackLIST then they should check themselves how this is handled in that case imho.

If you revert to accepting `Iterable` then disregard my previous comment concerning "every blacklist should be `set`".

Reply:
I prefer enforcing set because this way its clear to the user what the input should be and as long as the user follows this they cannot do wrong. I see that not allowing a list for Blacklist is a bit off. Iterable would be fine to me too.

Reply:
For reading and empty set we should still use set I think anyways